File: create-test-videos.js
#!/usr/bin/env node
/**
 * Create simple test video files using FFmpeg
 * This creates minimal MP4 files for testing the seed script
 */

const { execSync } = require('child_process');
const path = require('path');
const fs = require('fs');

const VIDEOS_DIR = path.join(__dirname, 'public', 'videos');

// Check if ffmpeg is available
function checkFFmpeg() {
  try {
    execSync('ffmpeg -version', { stdio: 'pipe' });
    return true;
  } catch (e) {
    return false;
  }
}

// Create a test video file
function createTestVideo(folderPath, filename, videoFormat = 'mp4') {
  const outputPath = path.join(folderPath, filename);
  
  if (fs.existsSync(outputPath)) {
    console.log(`⏭️  Skipping (already exists): ${filename}`);
    return true;
  }
  
  try {
    console.log(`📹 Creating: ${filename}`);
    
    // Create a 3-second test video with blue color and sine wave audio
    const cmd = `ffmpeg -f lavfi -i color=c=blue:s=640x480:d=3 -f lavfi -i sine=f=1000:d=3 -pix_fmt yuv420p -y "${outputPath}" 2>nul`;
    
    execSync(cmd, { stdio: 'pipe' });
    console.log(`✅ Created: ${filename}`);
    return true;
  } catch (error) {
    console.error(`❌ Error creating ${filename}:`, error.message);
    return false;
  }
}

function main() {
  console.log('🎬 Test Video File Creator\n');
  
  if (!checkFFmpeg()) {
    console.error('❌ ffmpeg not found!');
    console.error('\nPlease install ffmpeg:');
    console.error('  Windows: https://ffmpeg.org/download.html');
    console.error('  Mac: brew install ffmpeg');
    console.error('  Linux: sudo apt-get install ffmpeg');
    console.error('\nOr add video files manually to:');
    console.error(`  ${VIDEOS_DIR}`);
    process.exit(1);
  }
  
  if (!fs.existsSync(VIDEOS_DIR)) {
    console.error(`❌ Videos directory not found: ${VIDEOS_DIR}`);
    process.exit(1);
  }
  
  const videos = [
    { folder: 'action-movie-1', file: 'video.mp4' },
    { folder: 'tutorial-1', file: 'video.webm' },
  ];
  
  let created = 0;
  let skipped = 0;
  let failed = 0;
  
  console.log('Creating test videos...\n');
  
  videos.forEach(({ folder, file }) => {
    const folderPath = path.join(VIDEOS_DIR, folder);
    
    if (!fs.existsSync(folderPath)) {
      console.log(`⚠️  Folder not found: ${folder}`);
      failed++;
      return;
    }
    
    if (createTestVideo(folderPath, file)) {
      created++;
    } else {
      failed++;
    }
  });
  
  console.log(`\n📊 Summary:`);
  console.log(`   ✅ Created: ${created}`);
  console.log(`   ⏭️  Skipped: ${skipped}`);
  console.log(`   ❌ Failed: ${failed}`);
  
  if (created > 0) {
    console.log('\n✅ Done! Now run: npm run seed:videos');
  }
}

main();

File: create-test-videos.py
#!/usr/bin/env python3
"""
Create minimal test video files for seeding
Requires: ffmpeg installed and in PATH
"""

import os
import subprocess
import sys

VIDEOS_DIR = os.path.join(
    os.path.dirname(__file__),
    "public",
    "videos"
)

def create_test_video(folder_path, filename, duration=3):
    """Create a simple test video file using ffmpeg"""
    output_path = os.path.join(folder_path, filename)
    
    try:
        # Command to create a simple test video
        cmd = [
            "ffmpeg",
            "-f", "lavfi",
            "-i", "color=c=blue:s=640x480:d={}".format(duration),
            "-f", "lavfi",
            "-i", "sine=f=1000:d={}".format(duration),
            "-pix_fmt", "yuv420p",
            "-y",  # Overwrite output file
            output_path
        ]
        
        print(f"📹 Creating: {output_path}")
        subprocess.run(cmd, capture_output=True, check=True)
        print(f"✅ Created: {filename}")
        return True
    except subprocess.CalledProcessError as e:
        print(f"❌ Error creating {filename}: {e}")
        return False
    except FileNotFoundError:
        print("❌ ffmpeg not found. Please install ffmpeg:")
        print("   Windows: https://ffmpeg.org/download.html")
        print("   Mac: brew install ffmpeg")
        print("   Linux: sudo apt-get install ffmpeg")
        return False

def main():
    if not os.path.exists(VIDEOS_DIR):
        print(f"❌ Videos directory not found: {VIDEOS_DIR}")
        return
    
    print("🎬 Creating test video files...\n")
    
    # Define which folders need which video types
    videos_to_create = {
        "action-movie-1": "video.mp4",
        "tutorial-1": "video.webm",
    }
    
    created = 0
    failed = 0
    
    for folder, filename in videos_to_create.items():
        folder_path = os.path.join(VIDEOS_DIR, folder)
        
        if not os.path.exists(folder_path):
            print(f"⚠️  Folder not found: {folder}")
            continue
        
        if create_test_video(folder_path, filename):
            created += 1
        else:
            failed += 1
    
    print(f"\n📊 Summary:")
    print(f"   ✅ Created: {created}")
    print(f"   ❌ Failed: {failed}")
    
    if failed == 0 and created > 0:
        print("\n✅ Done! Now run: npm run seed:videos")

if __name__ == "__main__":
    main()
